src/main/groovy/io/seqera/wave/service/persistence/impl/WaveContainerRecordDeserializer.groovy

```diff
@@ -0,0 +1,72 @@
+
+        String id = node.hasNonNull("id") ? node.get("id").asText().replace("wave_request:", "") : null
+        User user = node.hasNonNull("user") ? parser.getCodec().treeToValue(node.get("user"), User.class) : null
+        Long workspaceId = node.hasNonNull("workspaceId") ? node.get("workspaceId").asLong() : null
+        String containerImage = node.hasNonNull("containerImage") ? node.get("containerImage").asText() : null
+        String containerFile = node.hasNonNull("containerFile") ? node.get("containerFile").asText() : null
+        ContainerConfig containerConfig = node.hasNonNull("containerConfig") ? parser.getCodec().treeToValue(node.get("containerConfig"), ContainerConfig.class) : null
+        String condaFile = node.hasNonNull("condaFile") ? node.get("condaFile").asText() : null
+        String platform = node.hasNonNull("platform") ? node.get("platform").asText() : null
+        String towerEndpoint = node.hasNonNull("towerEndpoint") ? node.get("towerEndpoint").asText() : null
+        String buildRepository = node.hasNonNull("buildRepository") ? node.get("buildRepository").asText() : null
+        String cacheRepository = node.hasNonNull("cacheRepository") ? node.get("cacheRepository").asText() : null
+        String fingerprint = node.hasNonNull("fingerprint") ? node.get("fingerprint").asText() : null
+        Instant timestamp = node.hasNonNull("timestamp") ? Instant.parse(node.get("timestamp").asText()) : null
+        String zoneId = node.hasNonNull("zoneId") ? node.get("zoneId").asText() : null
+        String ipAddress = node.hasNonNull("ipAddress") ? node.get("ipAddress").asText() : null
+        String sourceImage = node.hasNonNull("sourceImage") ? node.get("sourceImage").asText() : null
+        String sourceDigest = node.hasNonNull("sourceDigest") ? node.get("sourceDigest").asText() : null
+        String waveImage = node.hasNonNull("waveImage") ? node.get("waveImage").asText() : null
+        String waveDigest = node.hasNonNull("waveDigest") ? node.get("waveDigest").asText() : null
+        Instant expiration = node.hasNonNull("expiration") ? Instant.parse(node.get("expiration").asText()) : null
+        String buildId = node.hasNonNull("buildId") ? node.get("buildId").asText() : null
+        Boolean buildNew = node.hasNonNull("buildNew") ? node.get("buildNew").asBoolean() : null
+        Boolean freeze = node.hasNonNull("freeze") ? node.get("freeze").asBoolean() : null
+        String fusionVersion = node.hasNonNull("fusionVersion") ? node.get("fusionVersion").asText() : null
+
+        return new WaveContainerRecord(
+                id, user, workspaceId, containerImage, containerFile, containerConfig, condaFile, platform, towerEndpoint,
+                buildRepository, cacheRepository, fingerprint, timestamp, zoneId, ipAddress, sourceImage, sourceDigest,
+                waveImage, waveDigest, expiration, buildId, buildNew, freeze, fusionVersion
+        )
+    }
```

Comment:
Didn't know about this but too much code to maintain 😱

Reply:
I will see if i can reduce the code, mostly these are null checks to avoid NPE

Reply:
I have a patch using a regex, I'll push soon